tests/src/nodes/executor.rs

```diff
@@ -91,15 +94,20 @@ impl Executor {
         }
 
         config.storage.db_path = dir.path().join("db");
-        config
-            .da_sampling
-            .storage_adapter_settings
-            .blob_storage_directory = dir.path().to_owned();
-        config
-            .da_verifier
-            .storage_adapter_settings
-            .blob_storage_directory = dir.path().to_owned();
-        config.da_indexer.storage.blob_storage_directory = dir.path().to_owned();
+        dir.path().clone_into(
+            &mut config
+                .da_sampling
+                .storage_adapter_settings
+                .blob_storage_directory,
+        );
+        dir.path().clone_into(
+            &mut config
+                .da_verifier
+                .storage_adapter_settings
+                .blob_storage_directory,
+        );
+        dir.path()
+            .clone_into(&mut config.da_indexer.storage.blob_storage_directory);
```

Comment:
Why? Is there really a lint for this?
This is much more confusing than a single assignation...

Reply:
This is a `clippy::assigning_clones`, and it has do to with avoiding unnecessary allocations.

Reply:
I understand it is more optimal...but sincerly if clippy catches this the compiler should optimize it away, I wounder if it would :thinking: 
The point is that indeed, this is pedantic :joy:

Reply:
The assignation seems more optimized at first sight: https://rust.godbolt.org/z/War11carx

Reply:
Taken from the lint description:

> Custom `Clone::clone_from()` or `ToOwned::clone_into` implementations allow the objects to share resources and therefore avoid allocations.

I believe it depends on how the `Clone` trait is implemented, and I understand that some types are given the chance to implement `clone_from` and `clone_into` in a more performant way. So I think the lint as such makes sense, in this case, if you think we should revert to the old way, we can add an ignore, although I don't see a strong reason to do so.

Reply:
I do not see a strong reason to revert other than the one I gave. But I guess they know better. Also, I'm surprised it is the only place it occurs, though.